rest_api/rest_api/controller/health.py

```diff
@@ -0,0 +1,126 @@
+from typing import List
+
+import logging
+
+import os
+import pynvml
+import psutil
+
+from pydantic import BaseModel, Field
+
+from fastapi import FastAPI, APIRouter
+
+import haystack
+
+from rest_api.utils import get_app
+from rest_api.config import LOG_LEVEL
+
+logging.getLogger("haystack").setLevel(LOG_LEVEL)
+logger = logging.getLogger("haystack")
+
+
+router = APIRouter()
+app: FastAPI = get_app()
+
+
+class CPUUsage(BaseModel):
+    avg_1m: str = Field(..., description="Average CPU usage over the last 1 minute")
+    avg_5m: str = Field(..., description="Average CPU usage over the last 5 minutes")
+    avg_15m: str = Field(..., description="Average CPU usage over the last 15 minutes")
+
+
+class MemoryUsage(BaseModel):
+    total: str = Field(..., description="Total memory in megabytes")
+    used: str = Field(..., description="Used memory in megabytes")
+    percent: str = Field(..., description="Percentage of used memory")
+
+
+class GPUUsage(BaseModel):
+    memory_total: str = Field(..., description="Total GPU memory in megabytes")
+    memory_used: str = Field(..., description="Used GPU memory in megabytes")
+    memory_percent: str = Field(..., description="Percentage of used GPU memory")
+    kernel_usage: str = Field(..., description="GPU kernel usage in percent")
+
+
+class GPUInfo(BaseModel):
+    index: int = Field(..., description="GPU index")
+    usage: GPUUsage = Field(..., description="GPU usage details")
+
+
+class HaystackInfo(BaseModel):
+    version: str = Field(..., description="Haystack version")
+    cpu_usage: str = Field(..., description="REST API CPU usage details")
+    memory_usage: str = Field(..., description="REST API memory usage details")
+
+
+class SystemInfo(BaseModel):
+    cpu: CPUUsage = Field(..., description="System CPU usage details")
+    memory: MemoryUsage = Field(..., description="System memory usage details")
+    gpu_info: List[GPUInfo] = Field(default_factory=list, description="System GPU usage details")
```

Comment:
This might be a bit out of scope for the Haystack REST API... I'll ask @masci about his opinion. If there was a way to get the usage of GPU by the Haystack process (and add it to the `HaystackInfo` object) that would be super-cool tho :slightly_smiling_face:

Reply:
Do you think that maybe only the available memory? Usually RAM and GPU available are some concerns when doing NLP.

Reply:
I implemented the Haystack-specific GPU usage and removed some of the system-wide details. They are restricted now.